### PR TITLE
Fix/various uploader

### PR DESCRIPTION
**Pull request by J4bbi**
Fixes/features

- Tidies up search forms
- Gets rid of request way to pass information about multiple search fields, instead use Philip's approach
- adds person to work relationship data into the single relationship table
- bulk creates when importing and is much more efficient (https://github.com/culturesofknowledge/emlo-project/issues/145)
- accesses related data by annotating in the search query (this is efficient while the only 1 other table is queried this way, becomes incredibly slow if it's needed to annotate the third table) (https://github.com/culturesofknowledge/emlo-project/issues/139)
- correctly splits primary_name fields on import before checking for length
- update queryable work values of related items post-creation
- makes advanced search toggleable, moves general search toggle button to central control tab
- adds autocompletion to search-fields that support such functionality in old EMLO

### CODE REVIEW

File: location/views.py [hunk 1]
```diff
@@ -188,53 +188,47 @@ def sort_by_choices(self) -> list[tuple[str, str]]:
...
+
+        queryset = queryset.annotate(**annotate)
+
+        if queries:
+            queryset = queryset.filter(query_utils.all_queries_match(queries))
+
+        if sort_by := self.get_sort_by():
+            queryset = queryset.order_by(sort_by)
+
+        return queryset
```
Comment (philipkcl) [inline location/views.py:208]: suggest reuse super method 
```suggestion
        queryset = super().create_queryset_by_queries(model_class, queries)
        annotate = {'sent': Count('works',
                                  filter=Q(cofkworklocationmap__relationship_type=REL_TYPE_WAS_SENT_FROM)),
                    'recd': Count('works',
                                  filter=Q(cofkworklocationmap__relationship_type=REL_TYPE_WAS_SENT_TO)),
                    }
        annotate['all_works'] = annotate['sent'] + annotate['recd']
        queryset = queryset.annotate(**annotate)
        return queryset
```

File: work/templates/work/component/work_compact_search_fieldset.html [hunk 2]
```diff
@@ -70,15 +71,47 @@
...
+      '{{ subject }}',
+      {% endfor %}];
+
+     var catalogs_autocomplete = [{% for catalog in form.catalog_names %}
+      "{{ catalog|safe }}",
+      {% endfor %}];
+
+    </script>
+
+    <script src="/static/core/js/work_fieldset_autocomplete.js"></script>
```
Comment (philipkcl) [inline work/templates/work/component/work_compact_search_fieldset.html:115]: I did not have this file

File: location/views.py (hunk 1 above)
Comment (J4bbi) [reply location/views.py:208]: but what if the query is supposed to be sorted by sent, recd or all_works?

Comment (philipkcl) [reply location/views.py:208]: not sure will sorted sent work, if not working, we can extract.
```
        if queries:
            queryset = queryset.filter(query_utils.all_queries_match(queries))
        if sort_by := self.get_sort_by():
            queryset = queryset.order_by(sort_by)
```

**Pull request merged**